### ISSUE TITLE
cgi/escape does not define CGI.escapeHTML and other methods

### ISSUE THREAD

**Issue by jeremyevans** [jruby/jruby#4513]
### Environment

This is likely to be OS-independent, but:

```
$ jruby -v
jruby 9.1.7.0 (2.3.1) 2017-01-11 68056ae OpenJDK 64-Bit Server VM 25.112-b15 on 1.8.0_112-b15 +jit [OpenBSD-x86_64]
$ uname -a
OpenBSD testcurrent.bsa.ca.gov 6.0 GENERIC.MP#149 amd64
```

### Expected/Actual Behavior

`cgi/escape`, in addition to prepending to `CGI::Util`, also extends `CGI` itself: https://github.com/ruby/ruby/blob/ruby_2_3/ext/cgi/escape/escape.c#L104

Example of differences:

```
$ ruby23 -r cgi/escape -ve "p CGI.escapeHTML('<')"
"&lt;"
$ jruby -r cgi/escape -e "p CGI.escapeHTML('<')"
NoMethodError: undefined method `escapeHTML' for CGI:Class
  method_missing at org/jruby/RubyBasicObject.java:1653
          <main> at -e:1

$ ruby23 -r cgi/escape -e "p CGI.singleton_class.ancestors"
[#<Class:CGI>, CGI::Escape, #<Class:Object>, #<Class:BasicObject>, Class, Module, Object, Kernel, BasicObject]
$ jruby -r cgi/escape -e "p CGI.singleton_class.ancestors"
[#<Class:CGI>, #<Class:Object>, #<Class:BasicObject>, Class, Module, Object, Kernel, BasicObject]
```

**Comment by headius** [comment]
The CGI module we use is unmodified from C Ruby. Perhaps this is a 2.4 method?

**Comment by headius** [comment]
Oh nevermind, I see you tried Ruby 2.3. Strange that we don't have this. I think it might be a C ext they added and we never did.

**Comment by headius** [comment]
This was fixed in fd9515b5d7edaca80de426a1f43fa2af63a15a7b and will be part of our 2.4-compatible release, but it could be included in 9.1.8.0. @enebo?

**Comment by enebo** [comment]
@headius yeah I think it is a good idea.  My only question is will we add some 2.4ism if we cherry-pick this.  Maybe it does not matter all that much since escapeHTML will be what people are trying to hit and if we provide anything extra likely no one will be hitting it (and our 2.4 release will be pretty soon).

**Comment by headius** [comment]
@enebo I think it's pretty safe. There's only a couple functions in here...only took a couple hours to port.

**Comment by headius** [comment]
Actually I just realized this is already on master, but I think it's being wired up wrong.

**Comment by jeremyevans** [comment]
Thanks for the quick fix. I just want to point out that `cgi/escape` is in ruby 2.3, not just in 2.4 (see link to ruby repo in the original post), so this should be backported to 9.1.  Also, note that JRuby 9.1.7.0 already supported `cgi/escape`, it just didn't extend `CGI` with `CGI::Escape`.

**Comment by enebo** [comment]
@jeremyevans if you have the time to verify we landed all the missing bits or at least the bits you were trying to use (in this case we apparently already had them but just had them hooked up wrong).  

Sorry for the confusion about 2.3 vs 2.4.  I mostly meant if there was any change to cgi/escape in 2.4 and we ported for 2.4 something 2.4ish would come back to 2.3.  In any case, we did not have to contemplate the cherry-pick so it is all good.
